### DISCUSSION TITLE
TorBrowserDriver all parameters explained.

### DISCUSSION THREAD

**Discussion by RealSting** [webfp/tor-browser-selenium#181] [Q&A] [answered]
Hi mate,
please can you give some descriptions of these parameters ? :)

TorBrowserDriver(
**tbb_path**: str = "",
**tor_cfg**: int = cm.USE_RUNNING_TOR,
**tbb_fx_binary_path**: str = "",
**tbb_profile_path**: str = "",
**tbb_logfile_path**: str = "",
**tor_data_dir**: str = "",
**executable_path**: str = "geckodriver",
**pref_dict**: Any = {},
**socks_port**: Any | None = None,
**control_port**: Any | None = None,
**extensions**: Any = [],
**default_bridge_type**: str = "",
**headless**: bool = False,
**options**: Any | None = None,
**use_custom_profile**: bool = False,
**geckodriver_port**: int = 0)

**Comment by gunesacar** [comment, marked as answer]
It's a good idea to add annotations to code, but here are some brief descriptions:

`tbb_path`: path to Tor Browser Bundle (TBB) folder
`tor_cfg`: whether to start a new tor instance via Stem or use an already running (e.g. system tor) instance
`tbb_fx_binary_path`: path to Firefox binary inside TBB
`tbb_profile_path`: path to Firefox profile inside TBB
`tbb_logfile_path`: path to Tor Browser's log file
`tor_data_dir`: path to Tor's data directory
`executable_path`: path to geckodriver executable
`pref_dict`: dictionary of Firefox preferences to set
`socks_port`: SOCKS port to use
`control_port`: control port to use
`extensions`: list of paths to extensions to install
`default_bridge_type`: default Tor bridge type to use
`headless`: whether to launch Tor Browser in headless mode
`options`: Firefox options object
`use_custom_profile`: whether to launch from and *write to* a given profile
`geckodriver_port`: port to use for geckodriver
